src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -96,12 +97,113 @@ public static void Fill(Span<byte> data)
             RandomNumberGeneratorImplementation.FillSpan(data);
         }
 
+        public static int GetInt32(int fromInclusive, int toExclusive)
+        {
+            if (fromInclusive >= toExclusive)
+                throw new ArgumentException(SR.Argument_InvalidRandomRange);
+
+            // The total possible range is [0, 4,294,967,295).
+            long adjustedRange = (long)toExclusive - fromInclusive;
```

Comment:
This result fits into a uint, no need to make it a long.

Reply:
@GrabYourPitchforks Well, since ranges themselves can be negative it made sense to use long to allow subtracting a negative from a negative to just do the right thing. Are you suggesting

```csharp
uint adjustedRange = (uint)((long)toExclusive - fromInclusive);
```

Or do you want to avoid the long entirely?

Reply:
Oh I think you are proposing

```csharp
uint adjustedRange = (uint)toExclusive - (uint)fromInclusive;
```